common/subdevice-model.cpp

```diff
@@ -1745,14 +1745,19 @@ namespace rs2
     {
         bool is_d400 = s->supports(RS2_CAMERA_INFO_PRODUCT_LINE) ?
             std::string(s->get_info(RS2_CAMERA_INFO_PRODUCT_LINE)) == "D400" : false;
-
         std::string fw_version = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
             s->get_info(RS2_CAMERA_INFO_FIRMWARE_VERSION) : "";
-
         bool supported_fw = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
             is_upgradeable("05.11.12.0", fw_version) : false;
+        bool d400_on_chip_calib_supported = s->is<rs2::depth_sensor>() && is_d400 && supported_fw;
+
+        bool is_d500 = s->supports(RS2_CAMERA_INFO_PRODUCT_LINE) ?
+            std::string(s->get_info(RS2_CAMERA_INFO_PRODUCT_LINE)) == "D500" : false;
+        bool is_depth_sensor = s->supports(RS2_CAMERA_INFO_NAME) ?
+            std::string(s->get_info(RS2_CAMERA_INFO_NAME)) == "Stereo Module" : false;
+        bool d500_on_chip_calib_supported = is_depth_sensor && is_d500;
 
-        return s->is<rs2::depth_sensor>() && is_d400 && supported_fw;
+        return d400_on_chip_calib_supported || d500_on_chip_calib_supported;
         // TODO: Once auto-calib makes it into the API, switch to querying camera info
```

Comment:
How will you get it from camera info in the future?

Reply:
if you are talking about the comment - it was there before, and I don't know to what it refers - should I remove it?

Reply:
If we don't understand it yes please

Reply:
ok